spec/controllers/shoutouts_controller_spec.rb

```diff
@@ -0,0 +1,219 @@
+  describe 'POST #create' do
+    context 'with valid parameters' do
+      it 'creates a new shout-out with recipients of this shoutout' do
+        allow(controller).to receive(:current_user).and_return(user)
+        expect do
+          post :create, params: { shoutout: { user_id: user[:id],
+                                              time_period_id: time_period[:id],
+                                              rich_text: 'value2',
+                                              recipients: [user2.id, user3.id, user4.id] } }
+        end.to change(Shoutout, :count).by(1)
+        expect(ShoutoutRecipient.all.length).to be(3)
+      end
+
+      it 'not allow duplicate records' do
+        allow(controller).to receive(:current_user).and_return(user1)
+        expect do
+          post :create, params: { shoutout: { user_id: user1[:id],
+                                              time_period_id: time_period[:id],
+                                              rich_text: 'Similar message',
+                                              recipients: [user2.id, user3.id, user4.id] } }
+          post :create, params: { shoutout: { user_id: user1[:id],
+                                              time_period_id: time_period[:id],
+                                              rich_text: 'Similar message',
+                                              recipients: [user2.id, user3.id, user4.id] } }
+        end.to change(Shoutout, :count).by(1)
+        expect(ShoutoutRecipient.all.length).to be(3)
+      end
+    end
+
+    context 'with invalid parameters' do
+      it 'does not create a shout-out' do
+        allow(controller).to receive(:current_user).and_return(user)
+        expect do
+          post :create, params: { shoutout: { user_id: user[:id],
+                                              time_period_id: nil,
+                                              rich_text: 'value2',
+                                              recipients: [user2.id, user3.id, user4.id] } }
+        end.not_to change(Shoutout, :count)
+        expect(ShoutoutRecipient.all.length).to be(0)
+        expect do
+          post :create, params: { shoutout: { user_id: nil,
+                                              time_period_id: time_period[:id],
+                                              rich_text: 'value2',
+                                              recipients: [user2.id, user3.id, user4.id] } }
+        end.not_to change(Shoutout, :count)
+        expect(ShoutoutRecipient.all.length).to be(0)
+        expect do
+          post :create, params: { shoutout: { user_id: user[:id],
+                                              time_period_id: time_period[:id],
+                                              rich_text: nil,
+                                              recipients: [user2.id, user3.id, user4.id] } }
+        end.not_to change(Shoutout, :count)
+        expect(ShoutoutRecipient.all.length).to be(0)
+      end
+    end
```

Comment:
please divide it on three different tests

Reply:
Ok no problem, I divided it